### PR TITLE
Fixed the DXR samples so they can run on (I'm wheezing) Surface Book 3.

### PR DESCRIPTION
**Pull request by sherief**
Fixed the DXR samples so they can run on (I'm wheezing) Surface Book 3 and other MSHybrid systems.

### CODE REVIEW

**Comment by sherief** [comment]
TL;DR on the CLA, code is in the public domain glhf.

File: Samples/Desktop/D3D12Raytracing/src/D3D12RaytracingHelloWorld/DeviceResources.cpp [hunk 1]
```diff
@@ -597,14 +597,20 @@ void DeviceResources::MoveToNextFrame()
...
 
     ComPtr<IDXGIAdapter1> adapter;
-    for (UINT adapterID = 0; DXGI_ERROR_NOT_FOUND != m_dxgiFactory->EnumAdapters1(adapterID, &adapter); ++adapterID)
+    ComPtr<IDXGIFactory6> factory6;
+    HRESULT hr = m_dxgiFactory.As(&factory6);
+    if (FAILED(hr))
+    {
+        throw exception("DXGI 1.6 not supported");
+    }
+    for (UINT adapterID = 0; DXGI_ERROR_NOT_FOUND != factory6->EnumAdapterByGpuPreference(adapterID, DXGI_GPU_PREFERENCE_HIGH_PERFORMANCE, IID_PPV_ARGS(&adapter)); ++adapterID)
```
Comment (walbourn) [inline Samples/Desktop/D3D12Raytracing/src/D3D12RaytracingHelloWorld/DeviceResources.cpp:613]: There was a recent fix to make this a little more robust. Change this:

```
DXGI_ERROR_NOT_FOUND != m_dxgiFactory->EnumAdapters1(adapterID, &adapter); ++adapterID)
```

to

```
SUCCEEDED(m_dxgiFactory->EnumAdapters1(adapterID, &adapter); ++adapterID))
```

**Review by walbourn — CHANGES_REQUESTED** [review]

File: Samples/Desktop/D3D12Raytracing/src/D3D12RaytracingHelloWorld/DeviceResources.cpp (hunk 1 above)
Comment (sherief) [reply Samples/Desktop/D3D12Raytracing/src/D3D12RaytracingHelloWorld/DeviceResources.cpp:613]: That would definitely be an improvement, I'm only fixing one thing at (my own) time though, I don't think I will spend more time on this unless the core functionality of making it work on MSHybrid systems has issues.

**Comment by sherief** [comment]
I signed the CLA, should be merge-able as-is now.

**Comment by sherief** [comment]
Ping - any blockers on this?

**Comment by sebmerry** [comment]
I'll update adapter enumeration code for the DXR samples to bring it in line with the other samples.

**Comment by RobertChilewskiGit** [comment]
Have at Ye!

**Pull request merged**